src/intptrcast.rs

```diff
@@ -56,7 +56,7 @@ impl GlobalStateInner {
             int_to_ptr_map: Vec::default(),
             base_addr: FxHashMap::default(),
             exposed: FxHashSet::default(),
-            next_base_addr: STACK_ADDR,
+            next_base_addr: config.page_size.unwrap_or(128 * 1024),
```

Comment:
This is ultimately arbitrary, but it might still make sense to keep this the way it was before... can you pass `stack_addr` into this constructor?

Reply:
To clarify - should I just set a constant equal to 128k for this?

Reply:
No it should use `machine.stack_addr`, as it did before.

Reply:
Done